### PR TITLE
Add Excluded Tags to SQLite query using parameters

### LINKED ISSUE #2080: Creating a tag with a single quote in the name causes SQL crashes
**Describe the bug**

This is a bug as old as time. 

**To Reproduce**

Edit series metadata, add tag "Don't Show"
Edit user, block items with tag, add tag "Don't Show"
watch explosions.

**Expected behavior**

no crashes

**Logs**

**System (please complete the following information):**
 - OS: Docker
 - Browser: Chrome
 - Jellyfin Version: 10.4.1
 - Reverse proxy: no

### PR DESCRIPTION
**Pull request by neilsb**
**Changes**
Add Excluded Tags to SQLite query using parameters to avoid invalid SQL, or SQL injection type errors.

**Issues**
Fixes #2080

### CODE REVIEW

**Comment by neilsb** [comment]
Throughout ItemRepository (and others) there are similar cases where the SQL is built up using text replacement rather than SQLite parameters.  I feel all these should be updated, but probably best if a new issue is created for this as it's a fair bit of work.

File: Emby.Server.Implementations/Data/SqliteItemRepository.cs [hunk 1]
```diff
@@ -4593,10 +4593,19 @@ private List<string> GetWhereClauses(InternalItemsQuery query, IStatement statem
...
+                if (statement == null)
+                {
+                    int index = 0;
+                    whereClauses.Add("((select CleanValue from itemvalues where ItemId=Guid and Type=6 and cleanvalue in (" + string.Join(",", query.ExcludeInheritedTags.Select(t => paramName + index++)) + ")) is null)");
+                }
+                else
+                {
+                    for (int index = 0; index < query.ExcludeInheritedTags.Length; index++)
+                    {
+                        statement.TryBind(paramName + index, GetCleanValue(query.ExcludeInheritedTags[0]));
```
Comment (cvium) [inline Emby.Server.Implementations/Data/SqliteItemRepository.cs]: Should it not be `[index]`?

File: Emby.Server.Implementations/Data/SqliteItemRepository.cs [hunk 2]
```diff
@@ -4593,10 +4593,19 @@ private List<string> GetWhereClauses(InternalItemsQuery query, IStatement statem
...
             {
-                var tagValues = query.ExcludeInheritedTags.Select(i => "'" + GetCleanValue(i) + "'");
-                var tagValuesList = string.Join(",", tagValues);
-
-                whereClauses.Add("((select CleanValue from itemvalues where ItemId=Guid and Type=6 and cleanvalue in (" + tagValuesList + ")) is null)");
+                var paramName = "@ExcludeInheritedTags";
+                if (statement == null)
+                {
+                    int index = 0;
+                    whereClauses.Add("((select CleanValue from itemvalues where ItemId=Guid and Type=6 and cleanvalue in (" + string.Join(",", query.ExcludeInheritedTags.Select(t => paramName + index++)) + ")) is null)");
```
Comment (cvium) [inline Emby.Server.Implementations/Data/SqliteItemRepository.cs]: Could you break this up into multiple statements? It's a bit hard to follow just glancing at it.

File: Emby.Server.Implementations/Data/SqliteItemRepository.cs (hunk 1 above)
Comment (neilsb) [reply Emby.Server.Implementations/Data/SqliteItemRepository.cs]: Yes - sorry it should.  Correction pushed.

File: Emby.Server.Implementations/Data/SqliteItemRepository.cs (hunk 2 above)
Comment (neilsb) [reply Emby.Server.Implementations/Data/SqliteItemRepository.cs]: No problem - I've split this back out into a separate for loop to build up the list of parameters.  I'd just in-lined it to avoid creating a new `List<string>`.

Comment (cvium) [reply Emby.Server.Implementations/Data/SqliteItemRepository.cs]: Could also have been
```CS
string excludedTags = string.Join(",", query.ExcludeInheritedTags.Select(t => paramName + index++));
whereClauses.Add("((select CleanValue from itemvalues where ItemId=Guid and Type=6 and cleanvalue in (" + excludedTags + ")) is null)");
```

Comment (Bond-009) [reply Emby.Server.Implementations/Data/SqliteItemRepository.cs]: Use an array instead of a list, you know the size in advance

Comment (neilsb) [reply Emby.Server.Implementations/Data/SqliteItemRepository.cs]: I wasn't sure that just using the same functionality to generate it into a separate string was enough.  :)

Which way would be preferable and I can update?

Comment (Bond-009) [reply Emby.Server.Implementations/Data/SqliteItemRepository.cs]: Between these 3 options: LINQ, for loop + array or a StringBuilder, I would prefer the StringBuilder

Comment (neilsb) [reply Emby.Server.Implementations/Data/SqliteItemRepository.cs]: Pushed that change.

File: Emby.Server.Implementations/Data/SqliteItemRepository.cs [hunk 3]
```diff
@@ -4593,10 +4593,20 @@ private List<string> GetWhereClauses(InternalItemsQuery query, IStatement statem
...
+                {
+                    int index = 0;
+                    string excludedTags = string.Join(",", query.ExcludeInheritedTags.Select(t => paramName + index++));
+                    whereClauses.Add("((select CleanValue from itemvalues where ItemId=Guid and Type=6 and cleanvalue in (" + excludedTags + ")) is null)");
+                }
+                else
+                {
+                    for (int index = 0; index < query.ExcludeInheritedTags.Length; index++)
+                    {
+                        statement.TryBind(paramName + index, GetCleanValue(query.ExcludeInheritedTags[index]));
```
Comment (JustAMan) [inline Emby.Server.Implementations/Data/SqliteItemRepository.cs]: Do we still need to "clean" the value if we're properly binding it now? What that cleaning does anyway?

Comment (neilsb) [reply Emby.Server.Implementations/Data/SqliteItemRepository.cs]: The "cleaning" basically calls String.Normalize() as well as converting the tag to lowercase.

I suspect it should be left in to keep the search case insensitive.
